### DISCUSSION TITLE
git.iter_commits() cannot specify both a branch and commits between two revisions("reva....revb")

### DISCUSSION THREAD

**Discussion by socratesgorilla** [gitpython-developers/GitPython#1177] [Q&A] [answered]
What I mean is something identical to the function of, say, `git log upstream/master b59e00e...f79a12b`

This does not work for iter_commits, as rev only takes one argument. If there is a way to do this natively, it is not documented.

**Comment by Byron** [comment]
Did you try something like `len(list(r.iter_commits('HEAD~20..HEAD')))`? That worked as expected for me.
Maybe the documentation could be updated to explicitly mention that case if it works for you too -  a PR would then be very welcome.

**Comment by socratesgorilla** [comment]
It doesn't let you specify a branch, as far as I can tell.

**Comment by Byron** [comment, marked as answer]
The refspec `HEAD~20..HEAD` means all commits between the 20th ancestor of the currently checked out one (`HEAD~20`), and the currently checked out one (`HEAD`).
`HEAD` may as well be `branch`, meaning that `branch1..branch2` will work similarly.

Now it would be interesting to find out what happens if `branch1` is not an ancestor of `branch2`.

As this [capability is documented, too](https://github.com/gitpython-developers/GitPython/blob/4a1339a3d6751b2e7c125aa3195bdc872d45a887/git/repo/base.py#L519:L520), I don't see a need to keep this open any longer. If there is trouble with this nonetheless, please let me know in the comments.
